src/metrics/include/metrics/metrics.hpp

```diff
@@ -28,11 +29,57 @@
 namespace genny::metrics {
 
 /**
- * @namespace genny::metrics::v1 this namespace is private and only intended to be used by genny's
- * own internals. No types from the genny::metrics::v1 namespace should ever be typed directly into
- * the implementation of an actor.
+ * Class wrapping the logic involving metrics formats.
  */
-namespace v1 {
+class MetricsFormat {
+public:
+    enum class Format {
+        csv = 0,
+        cedar_csv = 1,
+        ftdc = 2,
+        csv_ftdc = 3,
+    };
```

Comment:
1. Convention is to have `kCamelCase`
2. No need to specify enum values

```c++
enum class Format {
    kCsv,
    kCedarCsv,
...
}
```

I'd also put a comment on what cedar csv is versus csv versus csv-ftdc 😄

Reply:
I don't think you need this if you just have `enum MetricsFormat` and you can add [a `struct NodeConvert` impl](https://github.com/mongodb/genny/blob/master/src/gennylib/include/gennylib/Node.hpp#L232-L249) so you can extract from the yaml

(but wait you're doing some logic in use_grpc and use_csv - do those need to be exposed outside of that one hpp file - can they just be `inline`d there?

Reply:
The use_grpc and use_csv get used in the registry, in the operationimpl, and in the default driver to make choices related to the new/old system. Ideally we wouldn't need these flags and in the future when we exorcise the old metrics system we could remove them.

Reply:
I ended up having a hard time getting the `NodeConvert` stuff to work right, so I ended up keeping it wrapped in the class and creating a constructor that takes a Node& so that extraction from yaml is more idiomatic. Let me know if you still think it should be an enum outside the class and I can ping you for questions about the `NodeConvert` struct.